core/src/main/java/aima/core/search/basic/uninformed/BidirectionalSearch.java

```diff
@@ -28,80 +29,120 @@
  * @author manthan
  */
 public class BidirectionalSearch<A, S> implements SearchForActionsBidirectionallyFunction<A, S> {
-    private BidirectionalActions<A> bidirectionalActions;
-    private List<A> fromInitialStatePartList = new ArrayList<>();
-    private List<A> fromGoalStatePartList = new ArrayList<>();
-    private Node<A, S> previousMeetingOfTwoFrontiers;
-    private Node<A, S> meetingOfTwoFrontiers;
-    private Node<A, S> nextNodeToBeEvaluated;
```

Comment:
this was wrong on many levels. why are there Queues created when the next node is always stored in an instance variable?